Proteomics/Protein/Protein.cs

```diff
@@ -225,11 +226,15 @@ public string GetEnsemblFastaHeader()
         /// <summary>
         /// Gets peptides for digestion of a protein
         /// </summary>
-        public IEnumerable<PeptideWithSetModifications> Digest(DigestionParams digestionParams, IEnumerable<Modification> allKnownFixedModifications,
+        public IEnumerable<PeptideWithSetModifications> Digest(DigestionParams digestionParams, List<Modification> allKnownFixedModifications,
             List<Modification> variableModifications, List<SilacLabel> silacLabels = null, (SilacLabel startLabel, SilacLabel endLabel)? turnoverLabels = null)
-        {
+        {            
             //can't be null
-            allKnownFixedModifications = allKnownFixedModifications ?? new List<Modification>();
+            allKnownFixedModifications = allKnownFixedModifications ?? new List<Modification>();            
+            if (digestionParams.Protease.CleavageMod!= null)
```

Comment:
this will add `digestionParams.Protease.CleavageMod` to `allKnownFixedModifications` every time `Digest` is called, meaning once for each protein being digested. change line 234 to include a `&& !allKnownFixedModifications.Contains(digestionParams.Protease.CleavageMod)` and then I think the PR is good to go

Reply:
change is made and comments have been added at some key places to hopefully help with understanding